src/app/core/services/auth.service.ts

```diff
@@ -74,30 +73,68 @@ export class AuthService {
     return this.authStateSource.getValue() === AuthState.Authenticated;
   }
 
-  setLoginStatusWithGithub(isLoggedIn: boolean) {
-    if (!isLoggedIn) {
-      this.electronService.clearCookies();
-    } else {
-      this.electronService.setCookie(this.githubUrl, this.githubDomain, this.githubLoginCacheName, 'yes');
-    }
-  }
-
   changeAuthState(newAuthState: AuthState) {
     if (newAuthState === AuthState.Authenticated) {
       const sessionId = `${Date.now()}-${uuid()}`;
       this.issueService.setSessionId(sessionId);
-      Logger.info(`Successfully authenticated with session: ${sessionId}`);
+      this.logger.info(`Successfully authenticated with session: ${sessionId}`);
     }
     this.authStateSource.next(newAuthState);
   }
 
   /**
    * Will start the Github OAuth web flow process.
-   * @param clearAuthState - A boolean to define whether to clear any auth cookies so prevent auto login.
    */
-  startOAuthProcess(clearAuthState: boolean = false) {
+  startOAuthProcess() {
     const githubRepoPermission = this.phaseService.githubRepoPermissionLevel();
     this.changeAuthState(AuthState.AwaitingAuthentication);
-    this.electronService.ipcRenderer.send('github-oauth', clearAuthState, githubRepoPermission);
+
+    if (this.electronService.isElectron()) {
+      this.electronService.sendIpcMessage('github-oauth', githubRepoPermission);
```

Comment:
Perhaps electron service can expose a method `sendOAuthStartMessage` instead of using `sendIpcMessage` directly here?

Reply:
Not sure about this. Currently, electron.service.ts acts as a 'safe' service for the web version to use. It prevents the web version from using undefined Electron functions which it has no access to.

For neatness, by exposing the `sendOAuthStartMessage` method means that we need to standardize all the low level methods in electron service to only expose the application's specific needs. For example, in this [line](https://github.com/CATcher-org/CATcher/pull/508/files#diff-0aaeb960028fa1ef5dbcc679364ac843587af5edef5746184171e17757c46ad7R58), we should also change it to a method like: `listenToElectronOAuthReply`. 

One disadvantage of doing so is that we would need to define a new method whenever a new requirement is needed to access the Ipc module.

Reply:
> For neatness, by exposing the sendOAuthStartMessage method means that we need to standardize all the low level methods in electron service to only expose the application's specific needs.

Yea, my suggestion would be that ElectronService should only expose the app's needs (reading a file, getting path of app directory, starting OAuth process) rather than exposing the ability to send / listen to messages between renderer process and main process (e.g. `sendIpcMessage` ).

> One disadvantage of doing so is that we would need to define a new method whenever a new requirement is needed to access the Ipc module.

Yes, this is true, though I think the electron service's usage will become more intuitive.
This is just a minor suggestion though. We can keep the existing style.